scaling_performance/optimizing_ephemeral_storage.adoc

```diff
@@ -0,0 +1,76 @@
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+*This page only applies if the Ephemeral Storage tech preview in
+{product-title} 3.10 has been enabled.  This is disabled by default;
+administrators may enable it via the instructions in
+xref:../install_config/configuring_ephemeral.adoc[Configuring for
+Ephemeral Storage].*
+
+Ephemeral storage is used by pods for their internal operation, such
+as for temporary files.  The lifetime of this ephemeral storage does
+not extend beyond the life of the individual pod, and this ephemeral
+storage cannot be shared across pods.
+
+Prior to {product-title} 3.10, ephemeral local storage was already
+exposed to pods via the container’s writable layer, logs directory,
+and EmptyDir volumes.  There are many issues related to the lack of
+local storage accounting and isolation, including:
+
+- Pods do not know how much local storage is available to them.
+
+- Pods cannot request "guaranteed" local storage.
+
+- Local storage is a "best effort" resource.
+
+- Pods can get evicted due to other pods filling up the local storage,
+after which no new pods will be admitted until sufficient storage has
+been reclaimed.
+
+Ephemeral storage is still exposed to pods in the same way, but there
+are new methods for implementing requests and limits on pods'
+consumption of ephemeral storage.
+
+Note that management of container logs apply only if you are using
+CRI-O as the container runtime and use file-based logging.
+
+It is important to understand that ephemeral storage is shared among
+all pods in the system, and that {product-title} does not provide any
+mechanism for guaranteeing any level of service beyond the requests
+and limits established by the administrator and users.  For example,
+ephemeral storage does not provide any guarantees of throughput, I/O
+operations/second, or any other measure of storage performance.
+
+== General Storage Guidelines
+
+A node's local storage can be broken into primary and secondary
+partitions.  Primary partitions are the only ones that can be used for
+ephemeral local storage.  There are two supported primary partitions:
+
+- *Root* -- This partition holds the kubelet’s root directory
+(`/var/lib/kubelet` by default) and `/var/log` directory.  This
+partition may be shared between user pods, OS and {product-title}
+system daemons.  This partition can be consumed by pods via EmptyDir
+volumes, container logs, image layers and container writable layers.
+{product-title} will manage shared access and isolation of this
+partition.
+
+- *Runtime* -- This is an optional partition which runtimes can use
+for overlay filesystems.  {product-title} will attempt to identify and
+provide shared access along with isolation to this partition.
+Container image layers and writable layers are stored here.  If the
+runtime partition exists, the `root` partition will not hold any image
+layer or writable layers.
+
+*TBD* Need description of how to set up runtime partition.
```

Comment:
remove this TBD

the kubelet discovers the configuration of the underlying container runtime (which on new installs is overlay by default)

Reply:
+1